documentation/ImplementingASemanticCheck.md

```diff
@@ -0,0 +1,813 @@
+<!--===- documentation/FrontEndTutorial.md 
```

Comment:
haven't decided the final name yet?

Reply:
Good catch!

And thanks for reading through this and providing feedback.